x-pack/plugins/observability_solution/observability_logs_explorer/public/state_machines/observability_logs_explorer/src/state_machine.ts

```diff
@@ -181,6 +201,11 @@ export const createObservabilityLogsExplorerStateMachine = ({
       createController: createController({ createLogsExplorerController }),
       initializeFromTimeFilterService: initializeFromTimeFilterService({ timeFilterService }),
       initializeFromUrl: initializeFromUrl({ urlStateStorageContainer, toastsService: toasts }),
+      initializeAllSelection: async (context) => {
+        const logSources = await logSourcesService.getLogSources();
+        const indices = logSources.map((logSource) => logSource.indexPattern).join(',');
+        return AllDatasetSelection.create({ indices });
+      },
```

Comment:
**nit**: surely non-blocking, but would be great if this could be a creator function as the other services do for readability purpose.